backend/buildeR.BLL/Services/SecretService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using VaultSharp;
+using VaultSharp.Core;
+using VaultSharp.V1.AuthMethods.Token;
+using VaultSharp.V1.Commons;
+
+namespace buildeR.BLL.Services
+{
+    public class SecretService
+    {
+        private readonly IVaultClient vaultClient;
+        private string path;
+
+        public SecretService()
+        {
+            vaultClient = CreateVaultClient();
+        }
+
+        public async Task<Dictionary<string, string>> CreateSecretsAsync(Dictionary<string, string> secrets, string path = null)
+        {
+            path = path ?? this.path;
```

Comment:
you can move this to a private method like `ValidatePath()` or `CheckPath()` to avoid duplicating code in methods